gensim/corpora/textcorpus.py

```diff
@@ -130,11 +130,14 @@ def sample_texts(self, n, seed=None, length=None):
             length = len(self)
 
         if not n <= length:
-            raise ValueError("n is larger than length of corpus.")
+
+            raise ValueError("n {0:d} is larger than length of corpus {1:d}.".format(n, length))
         if not 0 <= n:
-            raise ValueError("Negative sample size.")
+            raise ValueError("Negative sample size n {0:d}.".format(n))
 
+        seen_documents = 0
         for i, sample in enumerate(self.get_texts()):
+            seen_documents = i + 1
```

Comment:
Oh I think I see -- you can directly re-use the counter `i` from enumerate (just give it a better name).